File: apps/motoboy-pwa/src/userMessages.test.ts
import assert from "node:assert/strict";
import { test } from "node:test";
import { userFacingError } from "./userMessages";

test("keeps readable courier-facing errors", () => {
  assert.equal(
    userFacingError(new Error("Ative sua disponibilidade antes de aceitar corridas."), "Nao foi possivel atualizar agora."),
    "Ative sua disponibilidade antes de aceitar corridas.",
  );
});

test("hides technical backend and connection errors from courier UI", () => {
  const fallback = "Nao foi possivel atualizar agora.";

  assert.equal(userFacingError(new Error("Falha de comunicacao: 500"), fallback), fallback);
  assert.equal(userFacingError(new Error("connect ECONNREFUSED 127.0.0.1:3333"), fallback), fallback);
  assert.equal(userFacingError(new Error("NetworkError when attempting to fetch resource"), fallback), fallback);
  assert.equal(userFacingError(new Error("Request timeout after 12000ms"), fallback), fallback);
});

test("uses fallback for blank or unknown failures", () => {
  const fallback = "Nao foi possivel concluir agora.";

  assert.equal(userFacingError(new Error("   "), fallback), fallback);
  assert.equal(userFacingError("falha", fallback), fallback);
});

File: apps/motoboy/app/src/test/kotlin/com/drogsantoantonio/farmadelivery/presentation/ui/screens/UserMessagesTest.kt
package com.drogsantoantonio.farmadelivery.presentation.ui.screens

import kotlin.test.Test
import kotlin.test.assertEquals

class UserMessagesTest {
  @Test
  fun `keeps readable courier facing errors`() {
    assertEquals(
      "Ative sua disponibilidade antes de aceitar corridas.",
      courierFacingError(
        IllegalStateException("Ative sua disponibilidade antes de aceitar corridas."),
        "Nao foi possivel atualizar agora.",
      ),
    )
  }

  @Test
  fun `hides technical backend and connection errors`() {
    val fallback = "Nao foi possivel atualizar agora."

    assertEquals(fallback, courierFacingError(IllegalStateException("HTTP 500 Internal Server Error"), fallback))
    assertEquals(fallback, courierFacingError(IllegalStateException("java.net.UnknownHostException"), fallback))
    assertEquals(fallback, courierFacingError(IllegalStateException("connect ECONNREFUSED 127.0.0.1:3333"), fallback))
    assertEquals(fallback, courierFacingError(IllegalStateException("timeout after 12000ms"), fallback))
  }

  @Test
  fun `uses fallback for blank failures`() {
    assertEquals(
      "Nao foi possivel concluir agora.",
      courierFacingError(IllegalStateException("   "), "Nao foi possivel concluir agora."),
    )
  }
}
